src/main/java/io/odpf/firehose/factory/FirehoseConsumerFactory.java

```diff
@@ -86,8 +89,26 @@ public FirehoseConsumerFactory(KafkaConsumerConfig kafkaConsumerConfig, StatsDRe
      * @return FirehoseConsumer firehose consumer
      */
     public FirehoseConsumer buildConsumer() {
-
-        Filter filter = new MessageFilter(kafkaConsumerConfig, new Instrumentation(statsDReporter, MessageFilter.class));
+        FilterConfig filterConfig = ConfigFactory.create(FilterConfig.class, config);
+        Filter filter;
+        switch (filterConfig.getFilterEngine()) {
```

Comment:
as we are doing so much stuff for filters here, can we put this inside a private method?

Reply:
fixed